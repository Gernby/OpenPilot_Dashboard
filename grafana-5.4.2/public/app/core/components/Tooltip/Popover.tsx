import React from 'react';
import withTooltip from './withTooltip';
import { Target } from 'react-popper';

interface PopoverProps {
  tooltipSetState: (prevState: object) => void;
}

class Popover extends React.Component<PopoverProps, any> {
  constructor(props) {
    super(props);
    this.toggleTooltip = this.toggleTooltip.bind(this);
  }

  toggleTooltip() {
    const { tooltipSetState } = this.props;
    tooltipSetState(prevState => {
      return {
        ...prevState,
        show: !prevState.show,
      };
    });
  }

  render() {
    return (
      <Target className="popper__target" onClick={this.toggleTooltip}>
        {this.props.children}
      </Target>
    );
  }
}

export default withTooltip(Popover);
