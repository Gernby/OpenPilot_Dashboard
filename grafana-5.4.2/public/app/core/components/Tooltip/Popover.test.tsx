import React from 'react';
import renderer from 'react-test-renderer';
import Popover from './Popover';

describe('Popover', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Popover className="test-class" placement="auto" content="Popover text">
          <button>Button with Popover</button>
        </Popover>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
