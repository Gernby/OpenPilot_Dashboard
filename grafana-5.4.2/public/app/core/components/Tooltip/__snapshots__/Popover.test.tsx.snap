// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover renders correctly 1`] = `
<div
  className="popper__manager test-class"
>
  <div
    className="popper__target"
    onClick={[Function]}
  >
    <button>
      Button with Popover
    </button>
  </div>
</div>
`;
